juju1/state/migration_export.go

```diff
@@ -1317,10 +1317,15 @@ func (e *exporter) statusArgs(globalKey string) (description.StatusArgs, error)
 	return result, nil
 }
 
+const maxStatusHistoryEntries = 20
+
 func (e *exporter) statusHistoryArgs(globalKey string) []description.StatusArgs {
 	history := e.statusHistory[globalKey]
 	result := make([]description.StatusArgs, len(history))
 	e.logger.Debugf("found %d status history docs for %s", len(history), globalKey)
+	if len(history) > maxStatusHistoryEntries {
```

Comment:
Are we grabbing the right 20 here?

Reply:
Yup - it's coming in reverse-updated order, so the first 20 will be the most recent.